pkg/operator/controllers/machine/machine.go

```diff
@@ -75,37 +51,41 @@ func (r *MachineChecker) machineValid(ctx context.Context, machine *machinev1bet
 		return []error{fmt.Errorf("machine %s: failed to read provider spec: %T", machine.Name, o)}
 	}
 
+	// Validate VM size in machine provider spec
 	if !validate.VMSizeIsValid(api.VMSize(machineProviderSpec.VMSize), r.isLocalDevelopmentMode, isMaster) {
 		errs = append(errs, fmt.Errorf("machine %s: invalid VM size '%s'", machine.Name, machineProviderSpec.VMSize))
 	}
 
+	// Validate disk size in machine provider spec
 	if !isMaster && !validate.DiskSizeIsValid(int(machineProviderSpec.OSDisk.DiskSizeGB)) {
 		errs = append(errs, fmt.Errorf("machine %s: invalid disk size '%d'", machine.Name, machineProviderSpec.OSDisk.DiskSizeGB))
 	}
 
-	// to begin with, just check that the image publisher and offer are correct
+	// Validate image publisher and offer
 	if machineProviderSpec.Image.Publisher != "azureopenshift" || machineProviderSpec.Image.Offer != "aro4" {
 		errs = append(errs, fmt.Errorf("machine %s: invalid image '%v'", machine.Name, machineProviderSpec.Image))
 	}
 
+	// Validate managed identity
```

Comment:
I appreciate the effort of documenting the codebase, but some of these comments are unnecessary. Like this one: the code is straightforward and probably doesn't diserve a comment.

Reply:
I disagree. One-line comments do not negatively impact anyone's ability to read the code itself, if that is what one prefers to do. On the other hand, the absence of such comments can make it take much longer to understand 1. what is happening, and 2. what is intended to happen (they can differ, so a comment can help make that clear).

I would like to see more such comments in the codebase, so that it is easier for more people to jump in and understand enough to make contributions.

Reply:
I appreciate comments such as this when I'm skimming over the code, but if it's causing clutter we can remove the shorter ones.

Reply:
I'm very much in favour of having comments in the codebase. What I'm trying to say is - let's avoid code and comments like this:

```go
// Add two values
foo := baz + bar
```

I'm giving this as an example becuase error is saying "machine %s: invalid managedIdentity" and implies that the check has something to do with managedIdentity validation.

> 2. what is intended to happen (they can differ, so a comment can help make that clear).

For this we should use tests :)

Reply:
I agree, I've removed some unnecessary comments (ones with good error messages).